app/services/customers/create_service.rb

```diff
@@ -6,6 +6,10 @@ def create_from_api(organization:, params:)
       customer = organization.customers.find_or_initialize_by(external_id: params[:external_id])
       new_customer = customer.new_record?
 
+      unless valid_metadata_count?(metadata: params[:metadata])
+        return result.not_allowed_failure!(code: 'invalid_number_of_metadata')
```

Comment:
It suggest to render an unprocessable entity. As it's more a business logic validation than an "ability" to perform the action.
What do you think?

Reply:
👍 Changed